### ISSUE TITLE
Hyperlinks and cursors in layered views

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega-lite#3323]
In the new hyperlink and cursor functionality (which is a great addition - thanks!), when composing a layered view, each layer can have an independent `href` and `cursor` setting. While this makes sense for some designs, in others it does not.

A common use of layering is to have a text 'annotation layer'. However if the hyperlinking is applied only to the non-annotation layer(s), the text annotation on top 'blocks' the hyperlinking, including under whitespace:

```javascript
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {
    "values": [
      {"label": "Vega", "url": "https://vega.github.io/vega"},
      {"label": "Vega  -  Lite", "url": "https://vega.github.io/vega-lite"}
    ]
  },
  "layer": [
    {
      "mark": { "type": "circle", "cursor": "pointer" },
      "encoding": {
        "x": { "field": "label", "type": "nominal", "axis": null },
        "size": {  "value": 8000 },
        "color": { "field": "label", "type": "nominal", "legend": null},
        "href": { "field": "url","type": "nominal" }
      }
    },
    {
      "mark": "text",
      "encoding": {
        "x": { "field": "label","type": "nominal","axis": null },
        "text": { "field": "label", "type": "nominal" },
        "color": { "value": "white" },
        "size": { "value": 16 }
      }
    }
  ]
}
```

This can be fixed by adding the same `href` and `cursor` setting to the text layer, but I wonder if a better default would be to allow pass-through (i.e. union) of `cursor` and `href` settings with the ability to alter the default with `resolve`.

**Comment by kanitw** [comment]
Thanks for the suggestion. However, I think this suggested union by default is a bit too magical.  
How do you know when to apply href to another layer and when not to?

Unless we have a clear way to answer this question, this problem is easily solved by manually applying href and cursor to both layers.

**Comment by jwoLondon** [comment]
I agree it's no big problem in that one can always encode every layer with href and cursor. I realise you've closed the issue, so feel free to ignore the following, but I thought I should at least document my rationale.

The reasons for suggesting were prompted by what feels like an inconsistent behaviour with annotation layers. If you add an annotation layer over another that has some selection encoded, the annotation layer is effectively 'transparent' to the selection interaction. This seems like a good thing as it stops new layers from changing the interaction behaviour of existing ones. I see `href`s as being similar to a selection from a user's point of view, even if programmatically they are generated in a different way. Currently by default, a mouse click will be stopped by a non-interactive layer if it is to a URL, but not if it is for a data selection.

To answer @kanitw 's question, one model could be that a union is only applied to layers that have no hyperlink channel. This behaviour could be overridden or set with a (new) `href` option for `resolve`.

**Comment by domoritz** [comment]
I agree that it is a little inconsistent but it's a trade-off between making things consistent + adding something new that users have to learn and a slight inconsistency + an obvious fix.

**Comment by g3o2** [comment]
In vega-lite, there is no such thing as an "annotation layer", which happens to be in line with the grammar of graphics: text marks are just another way of representing data. What's more, any other mark than text can also be used for annotation.

The underlying vega could certainly benefit from an independent annotation facility, a currently [open feature request](https://github.com/vega/vega/issues/782). If it existed, vega-lite could build on that.

Meanwhile, the workaround consisting in duplicating href and cursor seems satisfying enough, for it still allows to solve the issue at hand.

**Comment by domoritz** [comment]
@g3o2 We're working with @starry97 on annotations. Stay tuned.
